internal/provider/org/role_resource.go

```diff
@@ -0,0 +1,425 @@
+	resp.TypeName = req.ProviderTypeName + "_org_role"
+}
+
+// Schema defines the schema for the resource.
+func (r *orgRoleResource) Schema(ctx context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "The Role resource allows you to manage a role in CloudAvenue.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The ID is a unique identifier for the role",
+			},
+			"name": schema.StringAttribute{
+				// Not ForceNew, to allow the resource name to be updated
+				Required:            true,
+				MarkdownDescription: "A Require name for the Role",
+			},
+			"description": schema.StringAttribute{
+				// Not ForceNew, to allow the resource name to be updated
+				Required:            true,
+				MarkdownDescription: "A require description for the role",
+			},
+			"bundle_key": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Key used for internationalization",
+			},
+			"read_only": schema.BoolAttribute{
+				Computed:            true,
+				MarkdownDescription: "Indicates if the role is read only",
+			},
+			"rights": schema.SetAttribute{
+				// Not ForceNew, to allow the resource name to be updated
+				Required:            true,
+				MarkdownDescription: "A require list of rights for the role",
+				ElementType:         types.StringType,
+			},
+		},
+	}
+}
+
+func (r *orgRoleResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.CloudAvenue)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *client.CloudAvenue, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *orgRoleResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Retrieve values from plan
+	var (
+		plan *orgRoleResourceModel
+		err  error
+	)
+
+	// Read the plan
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// role creation is accessible only in administrator API part
+	// (only administrator, organization administrator and Catalog author are allowed)
+	adminOrg, err := r.client.Vmware.GetAdminOrgByNameOrId(r.client.GetOrg())
+	if err != nil {
+		resp.Diagnostics.AddError("[role create] Error retrieving Org", err.Error())
+		return
+	}
+
+	// Check rights are valid
+	rights := make([]govcdtypes.OpenApiReference, 0)
+	for _, right := range plan.Rights.Elements() {
+		rg := strings.Trim(right.String(), "\"")
+		x, err := adminOrg.GetRightByName(rg)
```

Comment:
`rg` assignement is unnecessary 

```suggestion
		x, err := adminOrg.GetRightByName(strings.Trim(right.String(), "\"")
)
```